sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java

```diff
@@ -233,7 +233,11 @@ private PartitionWriter(TopicPartition tp, SinkRecord firstRecord) throws IOExce
               .orElseThrow(() -> new ConnectException("No local buffer directory configured"));
 
       File directory = new File(localBufferDirectory);
-      if (!directory.exists() && !directory.mkdirs()) {
+      int attempts = 3;
+      while (attempts > 0 && !directory.exists() && !directory.mkdirs()) {
+        attempts--;
```

Comment:
Does it really work without sleep?

Reply:
Sleep won't change anything here. By the time it fails, the directory already exists: the adjacent task on the same worker is faster with creating it.